protostar/compiler/project_compiler.py

```diff
@@ -104,6 +117,128 @@ def compile_contract_from_contract_identifier(
             )
         return self.compile_contract_from_contract_name(contract_identifier, config)
 
+    def compile_contract_to_sierra_from_contract_identifier(
```

Comment:
ProjectCompiler is Cairo0 entity, can we rename it to Cairo0Compiler instead and create separate entity for Cairo1?

Reply:
`compile_contract_to_sierra_from_contract_identifier` shouldn't be supported for cairo1, we should only support `ContractName`

Reply:
let's remove dead code please